src/app/api/participant/route.js

```diff
@@ -0,0 +1,38 @@
+import { getServerSession } from "next-auth/next";
+import { NextResponse } from "next/server";
+import { authOptions } from "../auth/[...nextauth]/route";
+import { db } from "../../../../firebase";
+import { doc, updateDoc } from "firebase/firestore";
+
+export async function POST(req) {
+  const res = NextResponse;
+  const { phone, major, age, school, grade, gender, shirt, diet } =
+    await req.json();
+  const session = await getServerSession(authOptions);
+
+  if (session) {
+    try {
+      await updateDoc(doc(db, "users", session.user.id), {
+        phone: phone,
+        major: major,
+        age: age,
+        school: school,
+        grade: grade,
+        gender: gender,
+        shirt: shirt,
+        diet: Object.entries(diet).map(([key, value]) => value && key),
```

Comment:
why is it value && key, key will always exist?

Reply:
if you just need the keys, then u can just do Object.keys and then check the value, cause this returns a 2d array